modin/experimental/batch/pipeline.py

```diff
@@ -0,0 +1,243 @@
+from modin.error_message import ErrorMessage
+from modin.core.dataframe.base.dataframe.utils import Axis
+from typing import Union, Callable, Optional
+from modin.config import NPartitions
+import ray
+import numpy as np
+from modin.distributed.dataframe.pandas.partitions import from_partitions
+
+
+class PandasQuery(object):
+    def __init__(
+        self,
+        func: Callable,
+        is_output: bool = False,
+        repartition_after: bool = False,
+        fan_out: bool = False,
+        pass_partition_id: bool = False,
+        reduce_fn: Optional[Callable] = None,
+    ):
+        self.func = func
+        self.is_output = is_output
+        self.oids = []
+        self.fan_out = fan_out
+        self.reptn = repartition_after
+        self.pass_num_ptn = pass_partition_id
+        self.operators = None
+        self.reduce_fn = reduce_fn
+
+    @classmethod
+    def print_options(cls):
+        """
+        Prints the kwargs that Query objects accept.
+        """
+        print(
+            """is_output : bool
+    Whether or not this query is an output query and should be passed both to the next query, and directly to postprocessing.
+repartition_after : bool
+    Whether or not to repartition after this query is computed. Currently, repartitioning is only supported if there is 1 partition prior to repartitioning.
+fan_out : bool
+    Whether or not to fan out this node. If True and only 1 partition is passed as input, the partition is replicated `num_partition` times, and the function is called on each. The `reduce_fn` must also be specified.
+pass_partition_id : bool
+    Whether or not to pass the numerical partition id to the query.
+reduce_fn : Calable
+    The reduce function to apply if `fan_out` is set to True. This takes the `num_partition` partitions that result from this query, and combines them into 1 partition."""
+        )
+
+
+class PandasQueryPipeline(object):
+    def __init__(self, df, axis: Union[int, Axis], num_partitions: int = 16):
+        self.df = df
+        self.axis = Axis(axis)
+        self.num_partitions = num_partitions
+        NPartitions.put(self.num_partitions)
+        NPartitions.get = lambda: self.num_partitions
+        self.unfinished = False
```

Comment:
what is `self.unfinished` for? I don't see any references to it.

Reply:
Good catch, thank you!